server/eventlog.go

```diff
@@ -88,17 +89,44 @@ func parsePCClientEventLog(rawEventLog []byte, pcrs *tpmpb.PCRs, loader Bootload
 	}, createGroupedError("failed to fully parse MachineState:", errors)
 }
 
-func parseCanonicalEventLog(rawCanonicalEventLog []byte, pcrs *tpmpb.PCRs) (*pb.MachineState, error) {
+// ParseCosCanonicalEventLogPCR takes in a raw COS CEL and PCR banks, validates and returns it's
+// COS states as parts of the MachineState.
+func ParseCosCanonicalEventLogPCR(cosEventLog []byte, pcrs *tpmpb.PCRs) (*pb.MachineState, error) {
+	tpm2Alg := tpm2.Algorithm(pcrs.GetHash())
+	cryptoHash, err := tpm2Alg.Hash()
+	if err != nil {
+		return nil, err
+	}
+	return getCosStateFromCEL(cosEventLog, cryptoHash, pcrs.GetPcrs(), cel.PcrTypeValue)
+}
+
+// ParseCosCanonicalEventLogRTMR takes in a raw COS CEL and a RTMR bank, validates and returns it's
+// COS states as parts of the MachineState.
+func ParseCosCanonicalEventLogRTMR(cosEventLog []byte, regs register.RTMRBank) (*pb.MachineState, error) {
```

Comment:
same comment about returning *pb.AttestedCosState.

comment: ParseCosCELRTMR takes an encoded COS CEL and RTMR bank, replays the CEL against the RTMRs, and returns the AttestedCosState